src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/service/impl/IdamElasticSearchServiceImpl.java

```diff
@@ -196,6 +203,79 @@ public ResponseEntity<Object> getIdamElasticSearchSyncFeed() {
+        LocalDateTime schedulerStartTime = now();
+        elinkDataIngestionSchedularAudit.auditSchedulerStatus(JUDICIAL_REF_DATA_ELINKS,
+            schedulerStartTime,
+            null,
+            RefDataElinksConstants.JobStatus.IN_PROGRESS.getStatus(), IDAMSEARCH);
+        Set<IdamResponse> judicialUsers = new HashSet<>();
+        List<UserProfile> userProfiles = userProfileRepository.fetchObjectIdFromCurrentDate();
+        userProfiles.forEach(userProfile -> {
+            Map<String, String> params = new HashMap<>();
+            params.put("size",String.valueOf(recordsPerPage));
+            params.put("query", idamFindQuery.concat(userProfile.getObjectId()));
+            log.debug("{}:: search elk query {}", loggingComponentName, params.get("query"));
+            int count = 0;
+            int totalCount = 0;
+            ResponseEntity<Object> responseEntity = null;
+            Response response = null;
+            try {
+                String bearerToken = "Bearer ".concat(getIdamBearerToken(schedulerStartTime));
+                response = idamFeignClient.getUserFeed(bearerToken, params);
+                logIdamResponses(response);
+                responseEntity = JsonFeignResponseUtil.toResponseEntity(response,
+                    new TypeReference<Set<IdamResponse>>() {
+                    });
+                if (response.status() == 200) {
+
+                    Set<IdamResponse> users = (Set<IdamResponse>) responseEntity.getBody();
+                    judicialUsers.addAll(users);
+
+                    List<String> headerCount = responseEntity.getHeaders().get("X-Total-Count");
+                    if (headerCount != null && !headerCount.isEmpty()
+                        && !headerCount.get(0).isEmpty()) {
+
+                        totalCount = Integer.parseInt(headerCount.get(0));
+                        log.debug("{}:: Header Records count from Idam :: " + totalCount, loggingComponentName);
+                    }
+
+                } else {
+                    log.error("{}:: Idam Search Service Failed :: ", loggingComponentName);
+                    throw new ElinksException(responseEntity.getStatusCode(), IDAM_ERROR_MESSAGE,
+                        IDAM_ERROR_MESSAGE);
+                }
+            } catch (Exception ex) {
+                //There is No header.
+                log.error("{}:: X-Total-Count header not return Idam Search Service::{}", loggingComponentName, ex);
+                elinkDataIngestionSchedularAudit.auditSchedulerStatus(JUDICIAL_REF_DATA_ELINKS,
+                    schedulerStartTime,
+                    now(),
+                    RefDataElinksConstants.JobStatus.FAILED.getStatus(), IDAMSEARCH);
+                throw new ElinksException(HttpStatus.valueOf(response.status()), ex.getMessage(),
+                    IDAM_ERROR_MESSAGE);
+            }
+        });
+
+        validateObjectIds(judicialUsers,schedulerStartTime);
```

Comment:
you are querying sidam with the object ids in the PRD. why do you need to validate and send email?